src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,25 @@
+import Popup from "./Popup";
+
+export class PopupWithConfirmation extends Popup {
+
+  constructor(popupSelector, handleConfirmDelete) {
+    super({ popupSelector });
+    this._popupButton = this._popupElement.querySelector(".modal__button");
+  }
+
+  setSubmit(action) {
+    this._handleSubmitCallback = action;
+  }
+
+  setEventListeners() {
+
+    super.setEventListeners();
+    this._popupButton.addEventListener("click", (evt) => {
+    evt.preventDefault();
+    this._handleSubmitCallback();
+    });
```

Comment:
You need to format the markup to show **indentation** and to make the code more readable. You can do that with `prettier` or a combination `SHIFT + ALT + F`


```js

  setEventListeners() {
    super.setEventListeners();
    this._popupButton.addEventListener("click", (evt) => {
      evt.preventDefault();
      this._handleSubmitCallback();
    });
  }
}

```